app/code/Magento/Catalog/Test/Unit/Model/CategoryLinkRepositoryTest.php

```diff
@@ -192,6 +249,83 @@ public function testDelete()
         $productMock->expects($this->once())->method('getId')->willReturn($productId);
         $categoryMock->expects($this->once())->method('setPostedProducts')->with([]);
         $categoryMock->expects($this->once())->method('save');
+
         $this->assertTrue($this->model->delete($this->productLinkMock));
     }
+
+    /**
+     * Delete by products skus
+     *
+     * @return void
+     */
+    public function testDeleteBySkus(): void
+    {
+        $categoryId = 42;
+        $productSku = 'testSku';
+        $productId = 55;
+        $productPositions = [55 => 1];
```

Comment:
please add case with use multiple product positions. Testing it with one product is good, but it's not the case that should be tested at first.